velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -16,124 +16,226 @@
 
 #include <utility>
 
+#include "velox/expression/DecodedArgs.h"
 #include "velox/expression/VectorFunction.h"
 #include "velox/expression/VectorWriters.h"
+#include "velox/functions/lib/Re2Functions.h"
 
 namespace facebook::velox::functions::sparksql {
 namespace {
 
-/// This class only implements the basic split version in which the pattern is a
-/// single character
-class SplitCharacter final : public exec::VectorFunction {
+/**
+ * split(string, delimiter[, limit]) -> array(varchar)
+ *
+ * Splits string on delimiter and returns an array of size at most limit.
+ * delimiter is a string representing regular expression.
+ * limit is an integer which controls the number of times the regex is applied.
+ * By default, limit is -1.
+ *
+ * If delimiter is not empty, then when limit > 0, the last element in
```

Comment:
Perhaps move the detailed description for `delimiter is not empty` to the splitInner function.